test/aleph/http_test.clj

```diff
@@ -541,7 +541,14 @@
         (d/chain' (fn [_] (s/close! body))))
     body))
 
+(defn init-stream-to-string-conversion! []
+  (bs/to-string (doto (s/stream 1) (s/put! "x") s/close!)))
```

Comment:
Maybe call it something like `force-memoization`?

Side note: I'm not really sure why Zach tried to delay it. Sure, you don't pay the cost of conversions you don't use, but most of them are just fns. I'd be curious to know the actual overhead.

Reply:
> Maybe call it something like `force-memoization`?

Sure, why not! Went with `force-stream-to-string-memoization!`

 
> Side note: I'm not really sure why Zach tried to delay it. Sure, you don't pay the cost of conversions you don't use, but most of them are just fns. I'd be curious to know the actual overhead.

My guess is that it's actually tricky to do this eagerly but still keep the conversion graph extensible. If you eagerly instantiate the converters (e.g. at the bottom of the `clj-commons.byte-streams` namespace), any third-party `def-conversion` declarations which would influence the optimal conversion paths would be ineffective. Alternatively, if you eagerly instantate the converters whenever a new conversion is declared, you'd have a lot of redundant instantiations during load time (factorial?) while all the stock conversions are being registered. I guess one way out of this pickle would be to provide an API function for eagerly instantiate the converters. This way, the user could call it at a point in their program when they know that all dependencies have been loaded. I'll file an issue over at https://github.com/clj-commons/byte-streams about it and will conduct some experiments to verify my hypotheses here.

Reply:
OK I just created https://github.com/clj-commons/byte-streams/issues/61 which picks up on this theme. Let's continue this thread over there :+1: